autoload/sj/go.vim

```diff
@@ -1,7 +1,6 @@
 function! sj#go#SplitImports()
   if getline('.') =~ '^import ".*"$'
-    s/^import \(".*"\)$/import (\r\1\r)/
-    normal! k==
+    s/^import \(".*"\)$/import (\r\t\1\r)/
```

Comment:
I see that you've removed the automatic indentation and manually added a tab? I thought go's convention was four spaces for indentation?

Either way, even if indentation varies, that's generally why I use `==` -- to rely on Vim's indentation choice (which is configurable by the user) rather than to enforce my own style of indentation.

What is the exact problem you're trying to solve with this change?

Reply:
> I thought go's convention was four spaces for indentation?

Nope, it's tabs.

> I use `==` -- to rely on Vim's indentation choice (which is configurable by the user) rather than to enforce my own style of indentation.

I think that would make sense for some other languages, but in Go there is the `gofmt` tool which automatically formats the code, and which has a near-universal community adoption. In other words, there aren't really any "styles of indentation" in Go, just the single canonical `gofmt` one.

That being said, could use `=` I suppose. Just seems to me that writing it correct the first time works better.

> What is the exact problem you're trying to solve with this change?

It gets written as:

```
import (
"fmt"
)
```

Rather than:

```
import (
    "fmt"
)
```

Reply:
> Nope, it's tabs.

Ah, right, it seems I even have that in my own config.

>  In other words, there aren't really any "styles of indentation" in Go, just the single canonical gofmt one.

That's fair, but in that case, it seems like this should be set in the Vim runtime files. Though, I do get they might not want to enforce it through the defaults -- we added this very recently in ruby. Even so, I think most Go developers would set this up themselves anyway.

> Just seems to me that just writing it correct seems to make more sense, instead of relying on autoformat tools?

I don't think it does, in the general case. Indentation can be surprisingly complicated, with a ton of edge cases. Applying any specific indentation in splitjoin is risky, because it means reimplementing logic that may or may not subtly change. This case is simple, but consider the thing that you yourself fixed with the var definition -- the indentation there is not a singe tab, but the amount that was already there plus a single tab. Even that is fixable, but this logic tends to accumulate and it's easier to simply add the newlines where they're needed and run `=` on the affected area.

I do have some hacks, whenever I *have* to add them for one reason or another, but ideally, I'd leave it to Vim to decide how to indent. And if there's a bug, it's the indent script that can and should be fixed, ideally.

In this particular case, I don't really know why this example doesn't work for you. It seems to indent correctly for me. Could you try pressing `==` on the "fmt" line here:

``` go
import (
"fmt"
)
```

Does it indent it correctly? If not, there might be some issue with your local go support. Are you using the built-in one, or maybe https://github.com/fatih/vim-go? Try pulling the master version?